packages/flutter_tools/lib/src/flx.dart

```diff
@@ -55,7 +56,38 @@ Future<int> createSnapshot({
     fs.file(depfilePath).parent.childFile('gen_snapshot.d').writeAsString('$depfilePath: $snapshotterPath\n');
   }
   args.add(mainPath);
-  return runCommandAndStreamOutput(args);
+
+  // TODO(cbracken): add checksumPath: '$depfilePath.checksums',
+  final BuildAction snapshotAction = new BuildAction(
+      inputPaths: await _readDepfile(depfilePath),
+      outputPaths: <String>[snapshotPath, depfilePath].toSet(),
+      buildFunction: (Set<File> inputs) async {
+        final int exitCode = await runCommandAndStreamOutput(args);
+        if (exitCode != 0)
+          throw new ToolExit('Failed to build snapshot');
```

Comment:
Agreed. I'll switch to `throwToolExit()` for consistency here, but I'll whip up a patch to migrate everything once this lands. Done.